src/app/ota_image_tool.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
+
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+"""
+Matter OTA (Over-the-air update) image utility.
+
+Usage examples:
+
+Creating OTA image file:
+./ota_image_tool.py create -v 0xDEAD -p 0xBEEF -vn 1 -vs "1.0" -da sha256 my-firmware.bin my-firmware.ota
+
+Showing OTA image file info:
+./ota_image_tool.py show my-firmware.ota
+"""
+
+import argparse
+import hashlib
+import os
+import struct
+import sys
+from enum import IntEnum
+
+sys.path.insert(0, os.path.join(
+    os.path.dirname(__file__), '../controller/python'))
+from chip.tlv import TLVReader, TLVWriter, uint  # noqa: E402
+
+HEADER_MAGIC = 0x1BEEF11E
+
+DIGEST_ALGORITHM_ID = dict(
+    sha256=1,
+    sha256_128=2,
+    sha256_120=3,
+    sha256_96=4,
+    sha256_64=5,
+    sha256_32=6,
+    sha384=7,
+    sha512=8,
+    sha3_224=9,
```

Comment:
Suggest warning if sha256_x (len < 256 bits) and sha3_224 are used, since:

> It is RECOMMENDED that a digest algorithm be chosen that has a minimum digest length of 256 bits, such as `sha-256` (ID 1 in the registry).